libraries/core_libs/consensus/include/dag/dag.hpp

```diff
@@ -251,6 +259,12 @@ class DagManager : public std::enable_shared_from_this<DagManager> {
   std::atomic<bool> stopped_ = true;
   std::thread block_worker_;
 
+  const uint32_t dag_expiry_limit_;  // Any non finalized dag block with a level smaller by
+                                     // dag_expiry_limit_ than the current period anchor level is considered
+                                     // expired and it should be ignored or removed from DAG
+  uint64_t dag_expiry_level_ = 0;    // Level below which dag blocks are considered expired, it's value is
```

Comment:
does not need to be thread safe?

Reply:
No, because every place where it is used is under DagManager::mutex_